polkadot/xcm/xcm-builder/src/weight.rs

```diff
@@ -138,15 +146,22 @@ where
 	M: Get<u32>,
 	Instruction<C>: xcm::latest::GetWeight<W>,
 {
-	fn weight_with_limit(message: &mut Xcm<C>, instrs_limit: &mut u32) -> Result<Weight, XcmError> {
+	fn weight_with_limit(
+		message: &mut Xcm<C>,
+		instrs_limit: &mut u32,
+	) -> Result<Weight, OutcomeError> {
 		let mut r: Weight = Weight::zero();
 		*instrs_limit = instrs_limit
 			.checked_sub(message.0.len() as u32)
-			.ok_or_else(|| XcmError::ExceedsStackLimit)?;
-		for instruction in message.0.iter_mut() {
+			.ok_or_else(|| OutcomeError { index: 0, error: XcmError::ExceedsStackLimit })?;
+		for (index, instruction) in message.0.iter_mut().enumerate() {
+			let index = index.try_into().unwrap_or(u8::MAX);
```

Comment:
> let index = index.try_into().unwrap_or(u8::MAX);

Well, yes, probably no other way, maybe also mentioned this limitation in docs for `index`.
But until `MAX_INSTRUCTIONS_TO_DECODE` does not change, we are safe here.

---

Maybe for correctness, we could add some integrity check for `MAX_INSTRUCTIONS_TO_DECODE`.

```
/// The maximal number of instructions in an XCM before decoding fails.
///
/// This is a deliberate limit - not a technical one.
pub const MAX_INSTRUCTIONS_TO_DECODE: u8 = 100;
```

Something like (if we had that alias `type InstructionIndex = u8;`):
```
assert!(MAX_INSTRUCTIONS_TO_DECODE < InstructionIndex::MAX)
```

But we're probably hitting the boundary of over-complication or over-engineering here.